### ISSUE TITLE
Tech Debt: Report ID generation

### ISSUE THREAD

**Issue by clediggins-usds** [CDCgov/prime-reportstream#5493]
### User Story:
As ReportStream, I want to be sure that I never have a Report ID overlap, to ensure data integrity.

### Description/Use Case
We use UUID.RandomUUID to generate report IDs. There is an extremely low chance of conflict; we should verify that the generated ID does not already exist in the database.

### Risks/Impacts/Considerations
* Potential risk of slowdown / greater database impact. Evaluate.

### Dev Notes:
* Edge case, but still possible.
* See UUID.ranomUUID() in Report.kt for where this is generated
* Will need to implement a data call to verify that a report id is not already in the database

### Acceptance Criteria
* We cannot have two of the same report ids in the database
* A conflict is detected, and a conflicting reportId will be regenerated until a non-conflicting one is found.

**Comment by JessicaWNava** [comment]
We decided as a team that we wanted to have the database set the uuid for the report id. You can do this via:

CREATE EXTENSION pgcrypto;
ALTER TABLE report_file ALTER COLUMN report_id SET DEFAULT gen_random_uuid();

However, it is only a default, there is not a way like with ints for it to auto assign them. We need a way to ensure that we only ever pass in null to the initial save. Another issue is that the save right now does not return the record, so if we are not passing in the value and the value is not returned on the save, we have no way of getting the id of the record which is needed for the rest of the processing. We may need to rethink how we are going to handle this.

**Comment by JessicaWNava** [comment]
Cle and I talked about this and since there is a lot of work that would create messier code for only marginal gain (only after generating 1 billion UUIDs every second for the next 100 years, the probability of creating just one duplicate would be about 50%) we are going to close this ticket and not do it.
